src/map/clif.cpp

```diff
@@ -12319,8 +12321,22 @@ void clif_parse_NpcBuyListSend( int fd, struct map_session_data* sd ){
 
 	if( sd->state.trading || !sd->npc_shopid )
 		result = e_purchase_result::PURCHASE_FAIL_MONEY;
-	else
-		result = npc_buylist( sd, n, (struct s_npc_buy_list*)p->items );
+	else{
+		std::vector<s_npc_buy_list> items;
```

Comment:
```suggestion
		std::vector<s_npc_buy_list> items = {};
```
Just following suite with how you declared it in `_buy`.